src/DynamicData/List/ChangeAwareList.cs

```diff
@@ -92,9 +92,9 @@ public IChangeSet<T> CaptureChanges()
                     var removed = returnValue.Unified().Select(u => u.Current);
                     returnValue = new ChangeSet<T> { new Change<T>(ListChangeReason.Clear, removed) };
                 }
-            }
 
-            ClearChanges();
+                ClearChanges();
```

Comment:
I think this should be inside the lock.  There was the slight possibility that a change could've happened after firing the changset, but before the lock was acquired for the clear.   Let me know if anyone disagrees with this change, as lock side effects can always be weird 8)